### DISCUSSION TITLE
Custom Layout: "use as default layout" options not working as expected

### DISCUSSION THREAD

**Discussion by stormwind84** [pimcore/pimcore#13448] [Q&A / Support] [answered]
Hello,
I would like to set a custom layout as default when creating and object, but if I mark "use as default layout" in the class settings, when creating a new object it always opens the master view (I have to select my custom layout manually). Tested on Pimcore 10.5.8 (in the demo seems to be the same). Has anyone experienced the same issue?
Many thanks,
Bob

**Comment by mattamon** [comment, marked as answer]
I did find the source of the problem.
https://github.com/pimcore/pimcore/blob/10.5/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php#L484

The currentLayoutId is set to 0 if you do not pick a layout yourself. When opening an object for the first time, the layoutId will always be 0.

The default layout therefore is never loaded. 

Would be nice if you could open a new issue please.

Thanks!

**Reply by stormwind84** [reply]
Yes, thanks @mattamon, will do it shortly.
